electrum/gui/qml/components/WalletMainView.qml

```diff
@@ -373,6 +373,45 @@ Item {
         }
     }
 
+    PIResolver {
+        id: piResolver
+        wallet: Daemon.currentWallet
+
+        onResolveError: (code, message) => {
+            var dialog = app.messageDialog.createObject(app, {
+                title: qsTr('Error'),
+                iconSource: Qt.resolvedUrl('../../icons/warning.png'),
+                text: message
+            })
+            dialog.open()
+        }
+
+        Component.onCompleted: {
+            piResolver.invoiceResolved.connect(invoiceParser.fromResolvedPaymentIdentifier)
+            piResolver.requestResolved.connect(requestDetails.fromResolvedPaymentIdentifier)
+        }
```

Comment:
> LGTM, but added one commit

^ this got lost in a rebase? https://github.com/spesmilo/electrum/commit/78b2e3df6c47ad5d5d119cbc2372e62807f372fd